harness/tests/experiment/fixtures/pytorch_onevar_model.py

```diff
@@ -148,23 +155,30 @@ def train_batch(
             "w_before": w_before,
             "w_after": w_after,
             "w_exp": w_exp,
+            "output": output,
         }
 
     @staticmethod
-    def check_batch_metrics(metrics: Dict[str, Any], batch_idx: int) -> None:
+    def check_batch_metrics(
+        metrics: Dict[str, Any],
+        batch_idx: int,
+        metric_keyname_pairs: Iterable[Tuple[str, str]] = (
+            ("loss", "loss_exp"),
+            ("w_after", "w_exp"),
+        ),
+        **kwargs,
```

Comment:
Let's avoid using kwargs like this, because it hurts the readability of the test.  If I read the function prototype of check_batch_metrics, I don't even know what kwargs is meant to get passed along to.  If you called it `atol` instead of `**kwargs`, it would accomplish the same purpose, only it would be more obvious how it was meant to be used, from reading the function definition.

Using `kwargs` makes a lot of sense when you are wrapping a function transparently or something, where the args the function takes are opaque to the wrapper.

Reply:
You're absolutely right.